### PR TITLE
Made folder names clickable

### LINKED ISSUE #1712: Open/Close folders by clicking on folder names
#### Nature of issue?

- Existing feature enhancement

#### Feature enhancement details:

Currently the contents of a folder are shown/hidden by clicking on the arrow button provided on the left of the folder name. However, I think it would be more convenient (from user's perspective) to show/hide the folder contents by clicking on the entire div. What are your thoughts about this @catarak? Can I work on this?

### PR DESCRIPTION
**Pull request by SundeepChand**
Fixes #1712

I have verified that this pull request:

* [x] has no linting errors (`npm run lint`)
* [x] is from a uniquely-named feature branch and has been rebased on top of the latest `develop` branch. (If I was asked to make more changes, I have made sure to rebase onto `develop` then too)
* [x] is descriptively named and links to an issue number, i.e. `Fixes #123`

Hello @catarak, please review this.

### CODE REVIEW

**Comment by catarak** [comment]
Thanks for working on this! Upon review, I don't like the way that the file name stays focused after you click on it, and I've been thinking about what would be better:
- I don't like that when you click on the open/close folder button, it loses focus. This is because each button is being show/hidden on click. I guess the solution would be to keep the button, but instead show/hide the icon and change the aria text.
- On clicking the folder name, maybe the focus could be blurred? Or the focus should be the whole row.

**Comment by SundeepChand** [comment]
Hi @catarak, I've made the changes. But it seems like something is wrong with the commit history. PS: Somehow I've managed to fix the commit history to some extent.

File: client/modules/IDE/components/FileNode.jsx [hunk 1]
```diff
@@ -256,18 +274,15 @@ class FileNode extends React.Component {
...
-                  onClick={this.showFolderChildren}
-                  aria-label={t('FileNode.OpenFolderARIA')}
-                >
-                  <FolderRightIcon className="folder-right" focusable="false" aria-hidden="true" />
-                </button>
-                <button
-                  className="sidebar__file-item-open"
-                  onClick={this.hideFolderChildren}
-                  aria-label={t('FileNode.CloseFolderARIA')}
+                  className={folderButtonClass}
```
Comment (catarak) [inline client/modules/IDE/components/FileNode.jsx:300]: I think maybe this should have `focusable=false`? I don't like that when you tab through the sidebar, you can focus on the folder name, it feels superfluous for keyboard users when you can just click on the arrow to the left!

**Review by catarak — CHANGES_REQUESTED** [review]

**Comment by SundeepChand** [comment]
Hello @catarak, I've made the changes, however, the folder options icon does not show up unless, the folder is toggled open/close.
May I know what does [this](https://github.com/processing/p5.js-web-editor/blob/61a9cb4f637dca60dc12368a8be0de26be210b7a/client/styles/components/_sidebar.scss#L67) comment mean? In order to fix the problems with folder options icon, it makes sense to move the hover functionality into JS.

**Comment by SundeepChand** [comment]
Hello @catarak, please review the changes made so far.
But I'm facing a strange issue right now. The current folder structure is as shown:

folder_s

When I first open `sketch.js`, then **open & close** `nested_folder` and then open `test.js`, `test.js` displays the contents of `sketch.js`. Moreover, this issue shows up a lot of times, but not everytime.

**Comment by catarak** [comment]
@SundeepChand I noticed this bug too! I'm not currently able to reproduce it on the `develop` branch so I _think_ it might be related to this PR, but I would double check me.

**Comment by catarak** [comment]
> May I know what does [this](https://github.com/processing/p5.js-web-editor/blob/61a9cb4f637dca60dc12368a8be0de26be210b7a/client/styles/components/_sidebar.scss#L67) comment mean? In order to fix the problems with folder options icon, it makes sense to move the hover functionality into JS.

If you hover over a file that is in a folder, you'll notice that the folder is also highlighted. I tried to a while to figure out how to generate the SCSS style so this wouldn't happen, but I couldn't figure it out. I bet this would be solved by switching to styled components (see #1760).

**Comment by Ajaya1000** [comment]
@catarak I found out how to solve this.
Screenshot from 2021-03-18 16-46-19
gitub

I'm also working on issue #1760 . So if you can merge my this branch it'll be easier for me to reference the style while writing styled components.

**Comment by Ajaya1000** [comment]
@catarak  should I create another issue for the hover effect??

**Comment by raclim** [comment]
Thank you so much for taking the time to contribute to this issue—I'm sorry we couldn't get to your work in time! Since some time has passed I'm going to close this for now, but please feel free to reopen this or work on this again, thanks!

**Pull request closed without merging**